crates/accelerate/src/elide_permutations.rs

```diff
@@ -78,17 +75,26 @@ fn run(py: Python, dag: &mut DAGCircuit) -> PyResult<Option<(DAGCircuit, Vec<usi
                 }
                 _ => {
                     // General instruction
-                    let mut mapped_inst = inst.clone();
                     let qargs = dag.get_qargs(inst.qubits);
+                    let cargs = dag.get_cargs(inst.clbits);
                     let mapped_qargs: Vec<Qubit> = qargs
                         .iter()
                         .map(|q| q.0 as usize)
                         .map(|q| mapping[q])
                         .map(|q| Qubit(q.try_into().unwrap()))
                         .collect();
-                    let mapped_qubits = new_dag.set_qargs(&mapped_qargs);
-                    mapped_inst.qubits = mapped_qubits;
-                    new_dag.push_back(py, mapped_inst)?;
+                    let params = inst.params.clone();
```

Comment:
Is this still needed? Or can we directly do `inst.params.as_deref().cloned()` in the stuct initializer?

Reply:
Good catch, this is a remnant of my own failed attempts to clone `params` correctly. Fixed in [8209757](https://github.com/Qiskit/qiskit/pull/13094/commits/820975721a5894315640edc105114d3a58ac9d60).

Reply:
Sounds good. We should probably (in a separate PR) expose a method on `PackedInstruction` as well which just returns `self.params.as_deref()` for convenience.

Reply:
I think that exists: https://github.com/Qiskit/qiskit/blob/main/crates/circuit/src/packed_instruction.rs#L530

Reply:
Hmm. Perhaps what I'm suggesting would be more confusing, then. The `params_view` method returns a `&[Param]` slice, but what we'd need here is the owned representation of parameters stored by `DAGCircuit` (`Option<SmallVec<[Param; 3]>>`).

That makes me think we may want to just change `DAGCircuit::apply_operation_{back,front}` to take `&[Param]` instead of owned data. It'd be less efficient for callers that happen to already have owned parameters to move in during the apply, but it's certainly a cleaner interface and hides what is otherwise an implementation detail for how DAGs store parameters.